### PR TITLE
Replace 'p isa Windows' by Sys.iswindows(p) etc.

### PR DESCRIPTION
**Pull request by fingolfin**
[skip ci]

Again, it seems better to changes this now in one go than to deal with this changes trickle by trickle.

### CODE REVIEW

**Comment by fingolfin** [comment]
To be clear, ideally I'd like to apply this without triggering *any* package updates, neither the wrappers nor the build files should be regenerated. I guess I need `[skip ci]` for that?

**Comment by giordano** [comment]
Yes, that's `[skip ci]`

**Comment by fingolfin** [comment]
Note that this PR was made using some regex and global search and replace, so on the one hand, I am fairly certain that I made no individual per-file errors, but on the other hand of course there is a non-zero risk I made a systematic error and/or overlooked a corner case. Not sure how to test for that, though...

File: L/libint/build_tarballs.jl [hunk 1]
```diff
@@ -22,7 +22,7 @@ rm ${prefix}/lib/liblibint2-static.a
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = filter!(p -> !isa(p, Windows), supported_platforms())
+platforms = filter!(p -> !Sys.iswindows(p), supported_platforms())
```
Comment (giordano) [inline L/libint/build_tarballs.jl:25]:
```suggestion
platforms = filter!(!Sys.iswindows, supported_platforms())
```

File: L/libjulia/common.jl [hunk 2]
```diff
@@ -178,7 +178,7 @@ function configure(version)
     filter!(!=(Linux(:i686, libc=:musl)), platforms)
 
     # For now skip FreeBSD...
-    filter!(p->!(p isa FreeBSD), platforms)
+    filter!(p->!Sys.isfreebsd(p), platforms)
```
Comment (giordano) [inline L/libjulia/common.jl:181]:
```suggestion
    filter!(!Sys.isfreebsd, platforms)
```

File: L/libsingular_julia/build_tarballs.jl [hunk 3]
```diff
@@ -42,7 +42,7 @@ const platforms = expand_cxxstring_abis([
 ])
 
 #platforms = supported_platforms()
-#platforms = filter!(p -> !(p isa Windows), platforms)
+#platforms = filter!(p -> !Sys.iswindows(p), platforms)
```
Comment (giordano) [inline L/libsingular_julia/build_tarballs.jl:45]:
```suggestion
#platforms = filter!(!Sys.iswindows, platforms)
```

File: M/MPICH/build_tarballs.jl [hunk 4]
```diff
@@ -75,7 +75,7 @@ make -j${nproc}
 make install
 """
 
-platforms = expand_gfortran_versions(filter!(p -> !isa(p, Windows), supported_platforms()))
+platforms = expand_gfortran_versions(filter!(p -> !Sys.iswindows(p), supported_platforms()))
```
Comment (giordano) [inline M/MPICH/build_tarballs.jl:78]:
```suggestion
platforms = expand_gfortran_versions(filter!(!Sys.iswindows, supported_platforms()))
```

File: M/MUMPS/MUMPS@5/build_tarballs.jl [hunk 5]
```diff
@@ -90,7 +90,7 @@ cp include/* ${prefix}/include
 
 # OpenMPI and MPICH are not precompiled for Windows
 # SCALAPACK doesn't build on PowerPC
-platforms = expand_gfortran_versions(filter!(p -> !isa(p, Windows) && arch(p) != :powerpc64le, supported_platforms()))
+platforms = expand_gfortran_versions(filter!(p -> !Sys.iswindows(p) && arch(p) != :powerpc64le, supported_platforms()))
```
Comment (giordano) [inline M/MUMPS/MUMPS@5/build_tarballs.jl:93]:
```suggestion
platforms = expand_gfortran_versions(filter!(p -> !Sys.iswindows(p) && arch(p) != "powerpc64le", supported_platforms()))
```

File: M/MicrosoftMPI/build_tarballs.jl [hunk 6]
```diff
@@ -44,7 +44,7 @@ mkdir -p share/licenses/MicrosoftMPI
 mv *.txt *.rtf share/licenses/MicrosoftMPI
 """
 
-platforms = filter!(p -> isa(p, Windows), supported_platforms())
+platforms = filter!(p -> Sys.iswindows(p), supported_platforms())
```
Comment (giordano) [inline M/MicrosoftMPI/build_tarballs.jl:47]:
```suggestion
platforms = filter!(Sys.iswindows, supported_platforms())
```

File: N/nauty/build_tarballs.jl [hunk 7]
```diff
@@ -49,7 +49,7 @@ install_license COPYRIGHT
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = filter(p -> !(p isa Windows), supported_platforms())
+platforms = filter(p -> !Sys.iswindows(p), supported_platforms())
```
Comment (giordano) [inline N/nauty/build_tarballs.jl:52]:
```suggestion
platforms = filter(!Sys.iswindows, supported_platforms())
```

File: N/normaliz/build_tarballs.jl [hunk 8]
```diff
@@ -24,7 +24,7 @@ make install
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 # windows build would require MPIR instead of GMP for 'long long'
-platforms = filter(x->!isa(x,Windows),supported_platforms())
+platforms = filter(x->!Sys.iswindows(p), supported_platforms())
```
Comment (giordano) [inline N/normaliz/build_tarballs.jl:27]:
```suggestion
platforms = filter(!Sys.iswindows, supported_platforms())
```

File: O/OpenMPI/build_tarballs.jl [hunk 9]
```diff
@@ -24,7 +24,7 @@ make "${flags[@]}" install
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line.
 #platforms = supported_platforms()
-platforms = filter(p -> !isa(p, Windows), supported_platforms())
+platforms = filter(p -> !Sys.iswindows(p), supported_platforms())
```
Comment (giordano) [inline O/OpenMPI/build_tarballs.jl:27]:
```suggestion
platforms = filter(!Sys.iswindows, supported_platforms())
```

File: O/opusfile/build_tarballs.jl [hunk 10]
```diff
@@ -28,7 +28,7 @@ make install
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = filter!(p -> !(p isa FreeBSD), supported_platforms())
+platforms = filter!(p -> !Sys.isfreebsd(p), supported_platforms())
```
Comment (giordano) [inline O/opusfile/build_tarballs.jl:31]:
```suggestion
platforms = filter!(!Sys.isfreebsd, supported_platforms())
```

File: P/PARMETIS/build_tarballs.jl [hunk 11]
```diff
@@ -36,7 +36,7 @@ make install
 """
 
 # OpenMPI and MPICH are not precompiled for Windows
-platforms = filter!(p -> !isa(p, Windows), supported_platforms())
+platforms = filter!(p -> !Sys.iswindows(p), supported_platforms())
```
Comment (giordano) [inline P/PARMETIS/build_tarballs.jl:39]:
```suggestion
platforms = filter!(!Sys.iswindows, supported_platforms())
```

File: P/Python/build_tarballs.jl [hunk 12]
```diff
@@ -62,7 +62,7 @@ platforms = supported_platforms()
 # Disable windows for now, until we can sort through all of these patches
 # and choose the ones that we need:
 # https://github.com/msys2/MINGW-packages/tree/1e753359d9b55a46d9868c3e4a31ad674bf43596/mingw-w64-python3
-platforms = filter(p -> !isa(p, Windows), platforms)
+platforms = filter(p -> !Sys.iswindows(p), platforms)
```
Comment (giordano) [inline P/Python/build_tarballs.jl:65]:
```suggestion
filter!(!Sys.iswindows, platforms)
```

File: Q/QD/build_tarballs.jl [hunk 13]
```diff
@@ -40,7 +40,7 @@ fi
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = expand_cxxstring_abis(filter!(p -> !isa(p, Windows), supported_platforms()))
+platforms = expand_cxxstring_abis(filter!(p -> !Sys.iswindows(p), supported_platforms()))
```
Comment (giordano) [inline Q/QD/build_tarballs.jl:43]:
```suggestion
platforms = expand_cxxstring_abis(filter!(!Sys.iswindows, supported_platforms()))
```

File: S/SPIRV_LLVM_Translator/SPIRV_LLVM_Translator@8/build_tarballs.jl [hunk 14]
```diff
@@ -6,7 +6,7 @@ version = v"8.0"
 sources = [GitSource(repo, "20240c9e4284a839337a04fb6001d8bf6b26fb62")]
 
 # LLVM 8's add_llvm_library does not work on Windows
-platforms = filter(p -> !(p isa Windows), platforms)
+platforms = filter(p -> !Sys.iswindows(p), platforms)
```
Comment (giordano) [inline S/SPIRV_LLVM_Translator/SPIRV_LLVM_Translator@8/build_tarballs.jl:9]:
```suggestion
platforms = filter(!Sys.iswindows, platforms)
```

File: S/sdsl_lite/build_tarballs.jl [hunk 15]
```diff
@@ -23,7 +23,7 @@ sed -ri \
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = expand_cxxstring_abis(filter!(p->!isa(p,Windows), supported_platforms()))
+platforms = expand_cxxstring_abis(filter!(p->!Sys.iswindows(p), supported_platforms()))
```
Comment (giordano) [inline S/sdsl_lite/build_tarballs.jl:26]:
```suggestion
platforms = expand_cxxstring_abis(filter!(!Sys.iswindows, supported_platforms()))
```

**Review by giordano — COMMENTED** [review]
Thank you so much for this!

**Comment by vchuravy** [comment]
As a side note I feel this really shouldn't have used `Sys.iswindows`. `Sys` for me is about the currently running system and it is an unecessary namespace pun. Could just have been `BB.iswindows`.

**Comment by giordano** [comment]
Fair point, but it's probably a bit too late to undo the use of `Sys` for the platforms, since this was done in `Pkg`, not in `BinaryBuilder` directly
